src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs

```diff
@@ -182,13 +200,61 @@ protected override Task CreateVersionsAsync(CancellationToken cancellationToken)
             // Add disabled versions
             AddBlockedVersions(blockedVersions);
 
+            // Update the view and add the filter
+            VersionsView = new CollectionViewSource() { Source = Versions }.View;
+            VersionsView.Filter += VersionsFilter;
+
             SelectVersion();
 
             OnPropertyChanged(nameof(Versions));
+            OnPropertyChanged(nameof(VersionsView));
+
+            // If the selected version is not the first one on the list, the line above will change it to the first one
+            if (_nugetProjects.Any() && 
+                _nugetProjects.FirstOrDefault().ProjectStyle.Equals(ProjectModel.ProjectStyle.PackageReference) &&
+                SelectedVersion != FirstDisplayedVersion)
+            {
+                SelectedVersion = FirstDisplayedVersion;
+                UserInput = SelectedVersion?.ToString();
+            }
 
             return Task.CompletedTask;
         }
 
+        private bool VersionsFilter(object o)
+        {
+            var version = o as DisplayVersion;
+            // If the text is empty or is the insalled version we should show all the versions like if there where no filtering
+            if (string.IsNullOrEmpty(UserInput) || UserInput.Equals(FirstDisplayedVersion?.ToString(), StringComparison.OrdinalIgnoreCase)) return true;
```

Comment:
What about `IsNullOrWhitespace` ? What does "  " do in the combobox?

Reply:
I think we need to leave it as `IsNullOrEmpty`, because right now " " will show no versions available, if we change to `IsNullOrWhitespace` if the user types "   " we will show all the versions, this can lead to the user typing "   X.y.x" (that is an available version) showing no versions available because Im not trimming the input.

You think we should trim? If so, using `IsNullOrWhitespace` is the best way

Reply:
I'm not sure whether spaces showing all versions is better than spaces showing no versions.
Maybe we just leave it alone for now.

Ideally: Spaces inputs are ignored until some acceptable characters are entered.
eg, typing repeatedly the space key does not change what is shown in the combobox.

Once a user types any valid thing, then spaces could be treated as acceptable?
Like comma followed by a space: "[1.0.0, 2.99.99]" ? 

Or to be very simple, we could say no spaces are accepted at all.